lib/improver/percentile.py

```diff
@@ -119,7 +119,19 @@ def process(self, cube):
         n_valid_coords = sum([test_coord == coord.name()
                               for coord in cube.coords()
                               for test_coord in self.collapse_coord])
-
+        # Check whether collapse_coord is a list, so that percentile_
+        # over_ can be renamed easily. This uses the list to get the
+        # name of the percentile_over_ variable e.g. if the list is
+        # ['longitude', 'latitude'] the name will be
+        # 'percentile_over_longitude_latitude'
+        name = 'percentile_over'
+        if type(self.collapse_coord) is list:
+            i = 0
+            while i < n_collapse_coords:
+                name = name + '_' + self.collapse_coord[i]
+                i += 1
+        else:
+            name = name + '_' + self.collapse_coord
```

Comment:
As I understand it, you're doing this to construct the name of the percentile coordinate that's created by `cube.collapsed`, so that you know which one to rename?

Can I suggest an alternative using the `improver.utilities.cube_checker.find_percentile_coord` function in the block below:
```
percentile_coord = find_percentile_coord(result)
result.coord(percentile_coord).rename('percentile')
```
This gets rid of the `if / else` statement completely.  Would that work?

Reply:
Yes that is why the if/else statement is there. I didn't know find_percentile_coordinate existed, otherwise I would have used that originally! This is all changed now though.